### PR TITLE
Low level driver for Si7006-A20

### PR DESCRIPTION
**Pull request by erikcorry**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+``` toit
+import fixed_point show *
+import gpio
+import serial.protocols.i2c as i2c
+import si7006 show *
+
+main:
+  sda := gpio.Pin --input --output 17
+  scl := gpio.Pin --input --output 16
```
Comment (floitsch) [inline README.md]: i2c overwrites the `input` and `output` properties. (takes over).
```suggestion
  sda := gpio.Pin 17
  scl := gpio.Pin 16
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# SI7006
```
Comment (andersjohnsen) [inline README.md:1]: package name should be lower case?

File: src/si7006.toit [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  device_ /i2c.Device ::= ?
+  static I2C_ADDRESS ::= 0x40
+
+  constructor .device_:
+
+  static FIRMWARE_VERSIONS_ ::= {
+    0xff: "Version 1.0",
+    0x20: "Version 2.0",
+  }
+
```
Comment (andersjohnsen) [inline src/si7006.toit:36]: Add on/off even though they are empty.

File: src/si7006.toit [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+  static POLYNOMIAL_ ::= 0b0011_0001  // x**8 + x**5 + x**4 + 1.
+
+  /// A string describing the firmware version on the sensor.
+  firmware -> string:
+    // We can't use registers_.read_bytes because that assumes a 1-byte
```
Comment (andersjohnsen) [inline src/si7006.toit]: Move this comment to the constructor so it's not per-method?

File: src/si7006.toit [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+  bus := i2c.Bus --sda=sda --scl=scl --frequency = 100_000
+  sensor_device := bus.device Si7006A20.I2C_ADDRESS
+  driver := Si7006A20 sensor_device
+  print "Firmware: $driver.firmware"
+  print "Serial#: $(%016x driver.serial_number)"
+  print "Temperature: $(driver.temperature)C"
+  print "Humidity: $(driver.humidity)%"
+```
+*/
+class Si7006A20:
```
Comment (andersjohnsen) [inline src/si7006.toit:26]: IMO we should use default import rules, thus name as:

```
class Driver
```
```
import si7006

main:
  driver := si7006.Driver ...
```

File: src/si7006.toit [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+  temperature -> float:
+    code := get_measurement_ 0xf3
+    return ((175.72 * code) / 65536.0) - 46.85
+
+  /// The current relative humidity measured by the sensor in percent.
+  humidity -> float:
+    code := get_measurement_ 0xf5
+    return ((125.0 * code) / 65536.0) - 6.0
+
+crc_8_ input/ByteArray polynomial/int --initial/int=0 -> int:
```
Comment (andersjohnsen) [inline src/si7006.toit:117]: Should this be in lib/crypto?

File: src/si7006.toit [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+    while true:
+      sleep --ms=time - 1
+      catch:
+        bytes := device_.read 3
+        check_crc_ bytes [0, 1] 2
+        return BIG_ENDIAN.uint16 bytes 0
+      time *= 2
+
+  /// The current temperature of the sensor in degrees Celcius
+  temperature -> float:
```
Comment (andersjohnsen) [inline src/si7006.toit]: I think the other THPs use read_temperature/read_humidity. We should probably pick one of them.

File: src/si7006.toit [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+    check_crc_ bytes1 [0, 2] 3
+    check_crc_ bytes1 [0, 2, 4] 5
+    check_crc_ bytes1 [0, 2, 4, 6] 7
+    check_crc_ bytes2 [0, 1] 2
+    check_crc_ bytes2 [0, 1, 3, 4] 5
+    model := bytes2[0]
+    // 00 and 0xff are engineering samples.  Normally we expect this byte of the
+    // serial number to be 6.
+    if model != 0 and model != 0xff and model != 6: throw "Not a 7006 device"
+    return bytes1[0] << 56
```
Comment (andersjohnsen) [inline src/si7006.toit:71]: little/big endian and uint32 ?

**Review by andersjohnsen — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (floitsch) [reply README.md:1]: My suggestion is to automatically lower case the title of the README for the package name.
If someone thinks that's a bad idea, please let me know.

File: src/si7006.toit [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+
+
+/**
+Driver for Si7006-A20 I2C Humidity and Temperature Sensor
+
+Example of use:
+```
+main:
+  sda := gpio.Pin --input --output 17
+  scl := gpio.Pin --input --output 16
```
Comment (floitsch) [inline src/si7006.toit]:
```suggestion
  sda := gpio.Pin 17
  scl := gpio.Pin 16
```

File: src/si7006.toit [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+  sensor_device := bus.device Si7006A20.I2C_ADDRESS
+  driver := Si7006A20 sensor_device
+  print "Firmware: $driver.firmware"
+  print "Serial#: $(%016x driver.serial_number)"
+  print "Temperature: $(driver.temperature)C"
+  print "Humidity: $(driver.humidity)%"
+```
+*/
+class Si7006A20:
+  device_ /i2c.Device ::= ?
```
Comment (floitsch) [inline src/si7006.toit]: ?
```suggestion
  device_ /i2c.Device
```

**Review by floitsch — APPROVED** [review]
LGTM.

**Pull request merged**